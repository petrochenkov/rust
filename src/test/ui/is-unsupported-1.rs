// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
// file at the top-level directory of this distribution and at
// http://rust-lang.org/COPYRIGHT.
//
// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
// option. This file may not be copied, modified, or distributed
// except according to those terms.

fn main() {
    let my_opt = Some(15);
    match 0u8 {
        0u8 if my_opt is Some(x) => { //~ ERROR cannot assign in a pattern guard
            println!("{}", x);
        }
        _ => {}
    }
}
