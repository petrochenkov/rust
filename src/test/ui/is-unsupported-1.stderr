error[E0302]: cannot assign in a pattern guard
  --> $DIR/is-unsupported-1.rs:14:16
   |
14 |         0u8 if my_opt is Some(x) => { //~ ERROR cannot assign in a pattern guard
   |                ^^^^^^^^^^^^^^^^^ assignment in pattern guard

error: aborting due to previous error

